File: Data Structures & Algorithms/boats-to-save-people/submission-1.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        left = 0
        right = len(people) - 1
        boats = 0

        while left <= right:
        # Heaviest person always takes a boat
            if people[left] + people[right] <= limit:
               left += 1  # Lightest person can share the boat
            right -= 1  # Move to next heaviest
            boats += 1

        return boats

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    validPalindrome(s) {
        let l = 0;
        let r = s.length -1;
  

        while(l < r){
            if(s[l] !== s[r] ){
                return (this.isPalindrome(s.slice(0, l) + s.slice(l+1)) || this.isPalindrome(s.slice(0,r)+ s.slice(r+1)))
            }
            l++;
            r--;
        }
        return true;


    }

    isPalindrome(s){
        let left = 0;
        let right = s.length - 1;
        while(left < right){
            if(s[left]!== s[right]){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}
